source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -989,10 +989,20 @@ int ServerContextImpl::handleOcspStapling(SSL* ssl, void*) {
     // We avoid setting the OCSP response if the client didn't request it, but doing so is safe.
     RELEASE_ASSERT(cert_context.ocsp_response_,
                    "OCSP response must be present under OcspStapleAction::Staple");
-    auto& resp_bytes = cert_context.ocsp_response_->rawBytes();
-    int rc = SSL_set_tlsext_status_ocsp_resp(ssl, const_cast<unsigned char*>(resp_bytes.data()),
-                                             resp_bytes.size());
-    RELEASE_ASSERT(rc != 0, "Error setting ocsp response");
+    const std::vector<uint8_t>& raw_bytes = cert_context.ocsp_response_->rawBytes();
+    const std::size_t raw_bytes_size = raw_bytes.size();
+    unsigned char* raw_bytes_copy = static_cast<unsigned char *>(OPENSSL_memdup(raw_bytes.data(), raw_bytes_size));
+    if (raw_bytes_copy == nullptr) { 
+      ENVOY_LOG_EVERY_POW_2_MISC(error, "OPENSSL_memdup failure");
+      stats_.ocsp_staple_failed_.inc();
+      return SSL_TLSEXT_ERR_ALERT_FATAL;
+    }
+    if (SSL_set_tlsext_status_ocsp_resp(ssl, raw_bytes_copy, raw_bytes_size) == 0) {
+      ENVOY_LOG_EVERY_POW_2_MISC(error, "SSL_set_tlsext_status_ocsp_resp failure");
+      OPENSSL_free(raw_bytes_copy);
+      stats_.ocsp_staple_failed_.inc();
+      return SSL_TLSEXT_ERR_ALERT_FATAL;
+    }
```

Comment:
You're leaking the copy? Shouldn't it be freed regardless the result of `SSL_set_tlsext_status_ocsp_resp`?

Reply:
IIRC when `SSL_set_tlsext_status_ocsp_resp` succeeds, that also means we transferred ownership of the memory. It's not clearly documented but this is what I remember inferring from reading up on this.

Reply:
It looks like it never fails anyway. It just transfers onwership: https://github.com/openssl/openssl/blob/master/ssl/s3_lib.c#L3552-L3556
So, we're good.

Reply:
re: it never fails anyway
Right - thanks for pointing that out. So reading that, I think a more pragmatic approach then is to not go and attempt to handle this situation gracefully, but instead have a `RELEASE_ASSERT`. 
That way we'll learn about it soon when this situation changes (theoretically there's nothing that says the call couldn't start doing it in future versions), while avoiding code that for practical purposes should be considered dead.

Reply:
Addressed in c3b2e8e5d6926d33cd31341ebf82c47670841159 (added a `RELEASE_ASSERT` and removed the code that attempted to handle this)